machine-learning/04-openpbta_plot_LV.Rmd

```diff
@@ -0,0 +1,379 @@
+ggplot(lv_plot_df,
+       aes(x = molecular_subtype, 
+           y = LV_estimate, 
+           group = molecular_subtype, 
+           colour = molecular_subtype)) +
+  geom_boxplot(outlier.shape = NA) +
+  geom_jitter(width = 0.2, alpha = 0.5) +
+  theme_bw() +
+  colorblindr::scale_color_OkabeIto() +
+  labs(title = lv_to_plot) +
+  theme(plot.title = element_text(size = 15,
+                                  face = "bold",
+                                  hjust = 0.5))
+```
+
+Use the next chunks to further customize your plot.
+We might suggest starting with the x- and y-axis labels.
+
+```{r labs_help}
+?labs
+```
+
+```{r adventure_time}
+
+```
+
+### Adding p-values to a plot
+
+R is great for analyzing genomic data and for publication ready visualizations, but as language for statistical computing, it's also great for statistical analysis!
+
+Let's start by performing a pairwise Wilcoxon rank sum test.
+We can put that information onto the plot with the `ggsignif` package.
+
+```{r pairwise_wilcox}
+pairwise.wilcox.test(x = lv_plot_df$LV_estimate,
+                     g = lv_plot_df$molecular_subtype,
+                     p.adjust.method = "bonferroni")
+```
+
+To add the Bonferroni-adjusted p-values for SHH vs. the other subtypes to the plot, we should adjust the y-axis limits to leave room for the labels.
+
+```{r increase_ylim, live = TRUE}
+# Make room for the annotations
+ggplot(lv_plot_df,
+       aes(x = molecular_subtype, 
+           y = LV_estimate, 
+           group = molecular_subtype, 
+           colour = molecular_subtype)) +
+  geom_boxplot(outlier.shape = NA) +
+  geom_jitter(width = 0.2, alpha = 0.5) +
+  theme_bw() +
+  colorblindr::scale_color_OkabeIto() +
+  labs(title = lv_to_plot) +
+  theme(plot.title = element_text(size = 15,
+                                  face = "bold",
+                                  hjust = 0.5)) +
+  ylim(c(-1.75, 4))
+```
+
+And now we're ready to add the adjusted p-values to the plot with `ggsignif::geom_signif()`.
+
+```{r geom_signif}
+# Add the adjusted p-values
+ggplot(lv_plot_df,
+       aes(x = molecular_subtype, 
+           y = LV_estimate, 
+           group = molecular_subtype, 
+           colour = molecular_subtype)) +
+  geom_boxplot(outlier.shape = NA) +
+  geom_jitter(width = 0.2, alpha = 0.5) +
+  theme_bw() +
+  colorblindr::scale_color_OkabeIto() +
+  labs(title = lv_to_plot) +
+  theme(plot.title = element_text(size = 15,
+                                  face = "bold",
+                                  hjust = 0.5)) +
+  ylim(c(-1.75, 4)) +
+  # Note this is highly specific to this plot!
+  # This can not be applied to any other plot without significant modification!
+  geom_signif(y_position = c(3.5, 3.9, 3.7),
+              xmin = c("Group3", "Group4", "SHH"),
+              xmax  = c("SHH", "SHH", "WNT"),
+              annotations = c("0.00071", "9e-06", "0.30052"),
+              tip_length = 0,
+              color = "#000000")
+```
```

Comment:
I would probably just combine these two chunks. How much to adjust y axis is also highly specific!